In [46]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [47]:
accident = pd.read_csv('C:\\Users\\IT-LAB3-PC09\\DESKTOP101\\Dataset\\accident_data.csv')

In [48]:
accident


Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    05/06/2019  51.506187   
1       200701BS65737           Serious    02/07/2019  51.495029   
2       200701BS66127           Serious    26/08/2019  51.517715   
3       200701BS66128           Serious    16/08/2019  51.495478   
4       200701BS66837            Slight    03/09/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18/02/2022  57.374005   
660675  201091NM01881            Slight    21/02/2022  57.232273   
660676  201091NM01935            Slight    23/02/2022  57.585044   
660677  201091NM01964           Serious    23/02/2022  57.214898   
660678  201091NM02142           Serious    28/02/2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [49]:
accident.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [83]:
accident['Light_Conditions', 'Road_Surface_Conditions'].corr ['Road_Type']

KeyError: ('Light_Conditions', 'Road_Surface_Conditions')

In [50]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [51]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [52]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna('Unknown Surface Condition')
accident['Road_Type'] = accident['Road_Type'].fillna('Unknown Road Type')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('Unknown Weather Condition')


In [53]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

In [54]:
accident.dtypes


Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [55]:
accident['Accident_Severity'] = accident['Accident_Severity'].astype('category')
accident['Latitude'] = accident['Latitude'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category')
accident['District Area'] = accident['District Area'].astype('category')
accident['Longitude'] = accident['Longitude'].astype('category')
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area']= accident['Urban_or_Rural_Area'].astype('category')
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype('category')
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype('category')

In [56]:
accident.dtypes

Index                        object
Accident_Severity          category
Accident Date                object
Latitude                   category
Light_Conditions           category
District Area              category
Longitude                  category
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [57]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst = True, errors = 'coerce')

In [58]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayOfWeek'] = accident['Accident Date'].dt.dayofweek

In [59]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DayOfWeek                  0
dtype: int64

<h1>UNIVARIATE<h1>

<h1>1. What are the most common accident severities?</h1>

In [60]:
accident['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

In [61]:
accident['Accident_Severity'].count()

660679

<h1>Insight 1</h1>
<h2>Out of 660679 accident count, 85.34%(563801) or the majority of accidents are classified as 'Slight', the other 13.35%(88217) lead to 'Serious' injury and the other 1.31 percentage result in 'Fatal' injuries.</h2>

<h1>2. What Year have the highest accidental rate?</h1>

In [62]:
accident['Year'].count()

660679

In [63]:
accident['Year'].value_counts()

Year
2019    182115
2020    170591
2021    163554
2022    144419
Name: count, dtype: int64

<h1>Insight 2</h1>
<h2>with 27.56% accidental rate, 2019 have the highest accidental rate followed by 2020 with 25.82%, 2021 with 24.76%, and 2022 with 21.86%</h2>
<h2>The results shows that as the years goes by, the accident counts decreases.</h2>

<h1>3. What is the most common vehicle type involved in accidents in 2019?</h1>

In [64]:
df_2019 = accident[accident['Year'] == 2019]

In [65]:
df_2019['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      130304
Bus or coach (17 or more pass seats)      10483
Van / Goods 3.5 tonnes mgw or under        9303
Motorcycle over 500cc                      7686
Goods 7.5 tonnes mgw and over              6247
Motorcycle 125cc and under                 4313
Taxi/Private hire car                      4230
Motorcycle over 125cc and up to 500cc      2446
Motorcycle 50cc and under                  2136
Goods over 3.5t. and under 7.5t            2049
Other vehicle                              1588
Agricultural vehicle                        636
Minibus (8 - 16 passenger seats)            625
Pedal cycle                                  68
Ridden horse                                  1
Data missing or out of range                  0
Name: count, dtype: int64

<h1>Insight 3</h1>
<h2> In 2019, Cars are the most involved vehicle type in accidents, significantly more than motorcycles and trucks.</h2>

<h1>4. How many accidents occurred in urban vs. rural areas?<h1>

In [66]:
accident['Urban_or_Rural_Area'].value_counts()


Urban_or_Rural_Area
Urban          421678
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h1>Insight 4</h1>
<h2>More accidents occur in urban areas due to higher traffic density, but rural accidents tend to be more severe.</h2>

<h1>5. What is the distribution of accidents by road type?</h1>

In [67]:
accident['Road_Type'].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Unknown Road Type       4520
Name: count, dtype: int64

<h1>Insight 5</h1>
<h2>Accidents frequently happen on major roads, but intersections and highways see more severe accidents.</h2>

<h1>6. What are the most common light conditions during accidents?</h1>

In [68]:
accident['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

In [88]:
accident.groupby(['Light_Conditions'])['Accident_Severity'].value_counts().unstack()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\805905202.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby(['Light_Conditions'])['Accident_Severity'].value_counts().unstack()


Accident_Severity            Fatal  Serious  Slight
Light_Conditions                                   
Darkness - lighting unknown     68      794    5622
Darkness - lights lit         1860    19130  108345
Darkness - lights unlit         45      360    2138
Darkness - no lighting        1612     7174   28651
Daylight                      5076    60759  419045

<h1>Insight 6</h1>
<h2>Most accidents happen in daylight, but nighttime accidents tend to be more severe.</h2>

<h1>7. What are the most common weather conditions during accidents?</h1>


In [69]:
accident['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds           520885
Raining no high winds         79696
Other                         17150
Unknown Weather Condition     14128
Raining + high winds           9615
Fine + high winds              8554
Snowing no high winds          6238
Fog or mist                    3528
Snowing + high winds            885
Name: count, dtype: int64

<h1>Insight 7</h1>
<h2>Clear weather conditions contribute to most accidents, possibly due to higher speeds and traffic density.</h2>

<h1>8. What month has the highest number of accidents?</h1>

In [70]:
accident['Month'].value_counts()

Month
11    60424
10    59580
7     57445
6     56481
9     56455
5     56352
3     54086
8     53913
1     52872
12    51836
4     51744
2     49491
Name: count, dtype: int64

<h1>Insight 8</h1>
<h2>November may have higher accident rates due to holiday traffic.</h2>

<h1>9. What day of the week has the most accidents?</h1>

<h1>10. What is the average number of vehicles involved per accident?</h1>

In [71]:
accident['Number_of_Vehicles'].mean()

1.8312554205597575

<h1>Insight 10<h1>
<h2>Most accidents involve only one or two vehicles, with multi-vehicle crashes being less common</h2>

<h1>11. How does accident severity vary between urban and rural areas?<h1>

In [72]:
accident.groupby('Urban_or_Rural_Area')['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\1797685776.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Urban_or_Rural_Area')['Accident_Severity'].value_counts()


Urban_or_Rural_Area  Accident_Severity
Rural                Slight               196077
                     Serious               37312
                     Fatal                  5601
Unallocated          Slight                   10
                     Serious                   1
                     Fatal                     0
Urban                Slight               367714
                     Serious               50904
                     Fatal                  3060
Name: count, dtype: int64

<h1>Insight 11<h1>
<h2>Urban areas see more slight accidents, while rural areas have a higher percentage of fatal accidents.<h2>

<h1>12. How does accident severity relate to the number of vehicles involved?<h1>

In [73]:
accident.groupby('Accident_Severity')['Number_of_Vehicles'].mean()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\939145392.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Accident_Severity')['Number_of_Vehicles'].mean()


Accident_Severity
Fatal      1.786976
Serious    1.678327
Slight     1.855864
Name: Number_of_Vehicles, dtype: float64

<h1>Insight 12<h1>
<h2> Fatal accidents involve fewer vehicles, suggesting single-vehicle crashes are deadlier.<h2>

<h1>13. How does accident severity change with weather conditions?<h1>

In [74]:
accident.groupby('Weather_Conditions')['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\3360046027.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Weather_Conditions')['Accident_Severity'].value_counts()


Weather_Conditions         Accident_Severity
Fine + high winds          Slight                 7134
                           Serious                1245
                           Fatal                   175
Fine no high winds         Slight               441739
                           Serious               72046
                           Fatal                  7100
Fog or mist                Slight                 2963
                           Serious                 483
                           Fatal                    82
Other                      Slight                15184
                           Serious                1801
                           Fatal                   165
Raining + high winds       Slight                 8209
                           Serious                1261
                           Fatal                   145
Raining no high winds      Slight                69380
                           Serious                9468
                    

<h1>Insight 13<h1>
<h2>Bad weather conditions (fog, rain) increase accident severity.<h2>

<h1>14. What is the relationship between light conditions and accident severity?<h1>

In [75]:
accident.groupby('Light_Conditions')['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\4147790636.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Light_Conditions')['Accident_Severity'].value_counts()


Light_Conditions             Accident_Severity
Darkness - lighting unknown  Slight                 5622
                             Serious                 794
                             Fatal                    68
Darkness - lights lit        Slight               108345
                             Serious               19130
                             Fatal                  1860
Darkness - lights unlit      Slight                 2138
                             Serious                 360
                             Fatal                    45
Darkness - no lighting       Slight                28651
                             Serious                7174
                             Fatal                  1612
Daylight                     Slight               419045
                             Serious               60759
                             Fatal                  5076
Name: count, dtype: int64

<h1>Insight 14<h1>
<h2>Nighttime accidents have a higher chance of being fatal due to reduced visibility.<h2>

<h1>15. Do certain vehicle types have more severe accidents?<h1>

In [76]:
accident.groupby('Vehicle_Type')['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\1246519302.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Vehicle_Type')['Accident_Severity'].value_counts()


Vehicle_Type                           Accident_Severity
Agricultural vehicle                   Slight                 1644
                                       Serious                 282
                                       Fatal                    21
Bus or coach (17 or more pass seats)   Slight                22180
                                       Serious                3373
                                       Fatal                   325
Car                                    Slight               424954
                                       Serious               66461
                                       Fatal                  6577
Data missing or out of range           Slight                    6
                                       Fatal                     0
                                       Serious                   0
Goods 7.5 tonnes mgw and over          Slight                14770
                                       Serious                2321
     

<h1>Insight 15<h1>
<h2>Motorcycles have the highest fatality rate, while cars mostly experience slight accidents.<h2>

<h1>16. How do accident rates vary by road surface conditions??<h1>


In [77]:
accident.groupby('Road_Surface_Conditions')['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\2693293023.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Road_Surface_Conditions')['Accident_Severity'].value_counts()


Road_Surface_Conditions    Accident_Severity
Dry                        Slight               380395
                           Serious               61638
                           Fatal                  5788
Flood over 3cm. deep       Slight                  842
                           Serious                 152
                           Fatal                    23
Frost or ice               Slight                16317
                           Serious                2007
                           Fatal                   193
Snow                       Slight                 5290
                           Serious                 565
                           Fatal                    35
Unknown Surface Condition  Slight                  654
                           Serious                  70
                           Fatal                     2
Wet or damp                Slight               160303
                           Serious               23785
                    

<h1>Insight 16<h1>
<h2>Wet and icy roads contribute to more serious and fatal accidents.<h2>

<h1>17. Do more accidents happen on specific road types?<h1>

In [78]:
accident.groupby('Road_Type')['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\172238111.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Road_Type')['Accident_Severity'].value_counts()


Road_Type           Accident_Severity
Dual carriageway    Slight                85863
                    Serious               11746
                    Fatal                  1815
One way street      Slight                11809
                    Serious                1655
                    Fatal                    95
Roundabout          Slight                40185
                    Serious                3665
                    Fatal                   142
Single carriageway  Slight               415557
                    Serious               70059
                    Fatal                  6527
Slip road           Slight                 6381
                    Serious                 611
                    Fatal                    49
Unknown Road Type   Slight                 4006
                    Serious                 481
                    Fatal                    33
Name: count, dtype: int64

<h1>Insight 17<h1>
<h2>Highways have the most severe crashes due to higher speeds.<h2>

<h1>18. Is there a correlation between the number of casualties and accident severity?<h1>

In [79]:
accident.groupby('Accident_Severity')['Number_of_Casualties'].mean()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\2287956350.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby('Accident_Severity')['Number_of_Casualties'].mean()


Accident_Severity
Fatal      1.903129
Serious    1.467280
Slight     1.331402
Name: Number_of_Casualties, dtype: float64

<h1>Insight 18<h1>
<h2> Fatal accidents result in a higher number of casualties per accident.<h2>

<h1>19. Do accidents increase during nighttime in urban vs. rural areas?<h1>


In [80]:
accident.groupby(['Urban_or_Rural_Area', 'Light_Conditions']).size()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\3769034877.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby(['Urban_or_Rural_Area', 'Light_Conditions']).size()


Urban_or_Rural_Area  Light_Conditions           
Rural                Darkness - lighting unknown      2467
                     Darkness - lights lit           24695
                     Darkness - lights unlit           961
                     Darkness - no lighting          35517
                     Daylight                       175350
Unallocated          Darkness - lighting unknown         0
                     Darkness - lights lit               2
                     Darkness - lights unlit             0
                     Darkness - no lighting              0
                     Daylight                            9
Urban                Darkness - lighting unknown      4017
                     Darkness - lights lit          104638
                     Darkness - lights unlit          1582
                     Darkness - no lighting           1920
                     Daylight                       309521
dtype: int64

<h1>Insight 19<h1>
<h2>Rural areas see more nighttime accidents due to poor lighting and higher speeds.<h2>

<h1>20. Which months have more severe accidents??<h1>

In [81]:
accident.groupby('Month')['Accident_Severity'].value_counts()

Month  Accident_Severity
1      Slight               45451
       Serious               6681
       Fatal                  740
2      Slight               42628
       Serious               6246
       Fatal                  617
3      Slight               46287
       Serious               7118
       Fatal                  681
4      Slight               43711
       Serious               7336
       Fatal                  697
5      Slight               47871
       Serious               7721
       Fatal                  760
6      Slight               47892
       Serious               7916
       Fatal                  673
7      Slight               48829
       Serious               7863
       Fatal                  753
8      Slight               45580
       Serious               7532
       Fatal                  801
9      Slight               47745
       Serious               7987
       Fatal                  723
10     Slight               50843
       Serious         

<h1>Insight 20<h1>
<h2>Winter months have higher fatal accident rates, possibly due to bad weather.<h2>

<h1>MultiVariate<h1>

<h1>21. How do accident severity, vehicle type, and location relate?<h1>

In [82]:
accident.groupby(['Vehicle_Type', 'Urban_or_Rural_Area'])['Accident_Severity'].value_counts()

C:\Users\IT-LAB3-PC09\AppData\Local\Temp\ipykernel_12520\1548730414.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  accident.groupby(['Vehicle_Type', 'Urban_or_Rural_Area'])['Accident_Severity'].value_counts()


Vehicle_Type                         Urban_or_Rural_Area  Accident_Severity
Agricultural vehicle                 Rural                Slight                 540
                                                          Serious                121
                                                          Fatal                   14
                                     Unallocated          Fatal                    0
                                                          Serious                  0
                                                                               ...  
Van / Goods 3.5 tonnes mgw or under  Unallocated          Fatal                    0
                                                          Slight                   0
                                     Urban                Slight               19135
                                                          Serious               2707
                                                          Fatal           

<h1>Insight 21<h1>
<h2>Motorcycles in rural areas have the highest fatality rate.<h2>

In [85]:
accident.groupby(['Road_Surface_ural_Area'])['Accident_Severity'].value_counts()

KeyError: 'Road_Surface_ural_Area'